### PR TITLE
[REF] git-hook: py script compatibility

### PR DESCRIPTION
**Pull request by moylop260**
- [REF] git-hook: Add compatibility for python script
 - [REF] git-hook: Add install pylint-odoo plugin and version variable
 - [REF] test_pylint: Add compatibility for git-hook.
 - [REF] self_tests: Add git-hook tests

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you please check errors on Travis?

**Comment by moylop260** [comment]
Fixed

**Comment by moylop260** [comment]
The error was because in this PR there were no changes in modules.
Then the pylint just in modified modules returned 0 errors.
Then I added a `git add --all` to force changes in all modules (after `mv` command)

File: git/README.md [hunk 1]
```diff
@@ -17,14 +17,14 @@ For using it, you can choose from two options:
 Then, making `git init` inside a git repository, these files will be copied,
 and you will have these checks available for it.
 
-**IMPORTANT** Don't forget to install/update flake8, pep8 and pylint modules to
+**IMPORTANT** Don't forget to install/update flake8, pep8 and pylint-oca-plugin modules to
```
Comment (pedrobaeza) [inline git/README.md]: s/pylint-oca-plugin/oca-pylint-plugin

Comment (moylop260) [inline git/README.md]: Fixed!

**Comment by moylop260** [comment]
@pedrobaeza 
Do you have another comment?

**Comment by moylop260** [comment]
@dreispt 
You feedback is welcome

**Comment by moylop260** [comment]
FYI rebased

**Comment by moylop260** [comment]
ping

**Comment by pedrobaeza** [comment]
I have tested and I'm afraid it's not working as expected. These are the issues I have found:

* You need to copy also travis folder directly in .git, which makes that you can't put on share folder as new git repos default. You should link travis folder inside git directory and make the rest of the references to it, and make it clear of instructions that if you just simply copy, you will copy the link, not the folder itself, so you need to make an absolute link, or hardcopy the contents.
* I get one warning per module like this:
```
__openerp__.py:19: [W0104(pointless-statement), ] Statement seems to have no effect
```
  so it seems there's some instruction about ignoring __openerp__.py files that is missing.

**Comment by moylop260** [comment]
>  I get one warning per module like this:

Next PR fix it version empty case: https://github.com/OCA/maintainer-quality-tools/pull/246
(Your feedback is welcome)
Try with `VERSION=8.0 git commit`

Thanks for you revision!

**Comment by pedrobaeza** [comment]
Independent from #246, here we should set an automatic version variable check, based on the name of the branch, because it very tedious to put version environment variable each time you make a commit.

What about the other question?

**Comment by moylop260** [comment]
Ok, branches startswith version is a cool idea.
I will change this script.

The other question, I dont know if you want help me (my english cae)...

**Comment by pedrobaeza** [comment]
The other question is also a problem, that's why I tell you to solve it. Can you read it again and tell me if something you don't understand?

Thanks.

**Comment by moylop260** [comment]
Ok
(I will read again, I saw a documentation case in first view... sorry)

**Comment by pedrobaeza** [comment]
@moylop260, any advance on this?

**Comment by moylop260** [comment]
@pedrobaeza 
Now suport `get version from branch name`
And it works.
But I don't understand fully the comment https://github.com/OCA/maintainer-quality-tools/pull/242#issuecomment-141835921 yet

**Comment by pedrobaeza** [comment]
Explain in spanish:

"Hace falta copiar también la carpeta travis directamente en la carpeta .git de cada repositorio a comprobar, lo que hace que no se pueda colocar en la carpeta compartida (/usr/share/git-core/templates/hooks/) para que se utilice por defecto para los nuevos repositorios git."

**Comment by moylop260** [comment]
@pedrobaeza 
I use next configuration and works fine: https://asciinema.org/a/24642
Without copy travis folder in each repository.

Can you send me your commands executed to check it?

**Comment by moylop260** [comment]
@pedrobaeza 
I changed the README file to use symbolic link process.

**Comment by moylop260** [comment]
@pedrobaeza 
FYI we are using this script in our laptops work fine!

File: git/README.md [hunk 2]
```diff
@@ -7,24 +7,53 @@ OCA maintainer quality tools.
...
+   ```
+
+2. Create symbolic link of files to git template directory.
+
+  In Ubuntu, it's located in:
+
+     /usr/share/git-core/templates/hooks/
+
+
+  In MAC, it's located in:
```
Comment (osvalr) [inline git/README.md]: In OS X

**Comment by yaniaular** [comment]
:+1:  I am using it. It works perfect, i like it.

File: git/README.md (hunk 2 above)
Comment (moylop260) [inline git/README.md]: @osvalr 
Fixed!
Thanks

File: git/README.md [hunk 3]
```diff
@@ -7,24 +7,53 @@ OCA maintainer quality tools.
...
+
+    Installed from MAC-port
+
+        /opt/local/share/git-core/templates/hooks/
 
-   /usr/share/git-core/templates/hooks/
 
 Then, making `git init` inside a git repository, these files will be copied,
```
Comment (osvalr) [inline git/README.md]: I would recommend to make an bash-based alias to do that as the same command used in https://github.com/OCA/maintainer-quality-tools/pull/242/files#diff-eee548fe380dc59c94cc87563f0d454cR15 in the main directory as the repo is located, i.e.

```bash
alias install_git_hook='[[ -d .git ]] && ls -sf ${HOME}/lint-hook/git/pre-commit .git/hooks/pre-commit || echo "failed"'
```
IMHO this a user can be a handy alias to use this when is required in a existing repo

Comment (moylop260) [inline git/README.md]: @osvalr 
Good comment with a `install hook`.
Now is too much for this PR that have a focus: `compatibility with python migration`
But you can create other one (progressive maybe) with this feature.

**Comment by osvalr** [comment]
I am using it, this makes daily work less time-consuming than wait for CI to check. :+1:

File: git/README.md [hunk 4]
```diff
@@ -7,24 +7,53 @@ OCA maintainer quality tools.
...
+**IMPORTANT** Don't forget to install/update flake8, pep8 and pylint-odoo modules to
 have the same checks as Travis. Don't use distribution packages, as they are 
 outdated. Use pip ones:
-```
-sudo pip install flake8 --upgrade
-sudo pip install pep8 --upgrade
-sudo pip install pylint --upgrade
+
+```bash
+# pip install --upgrade flake8 pep8
```
Comment (pedrobaeza) [inline git/README.md]: Here you have to include click library also. We might need to replace it by the standard parser...

File: git/README.md [hunk 5]
```diff
@@ -7,24 +7,53 @@ OCA maintainer quality tools.
...
+$ git clone https://github.com/OCA/pylint-odoo
+$ cd pylint-odoo
+# ./install.sh
```
Comment (pedrobaeza) [inline git/README.md]: Why install.sh? This doesn't exists

File: git/README.md (hunk 4 above)
Comment (pedrobaeza) [inline git/README.md]: And also pylint-odoo...

**Comment by pedrobaeza** [comment]
I merge and change instructions afterwards

File: git/README.md (hunk 4 above)
Comment (moylop260) [inline git/README.md]: Pylint-odoo requirements.txt have click then is installed as dependency.

File: git/README.md (hunk 5 above)
Comment (moylop260) [inline git/README.md]: In pylint-odoo projects exists install.sh because we need nodejs packages.

File: git/README.md (hunk 4 above)
Comment (pedrobaeza) [inline git/README.md]: Yeah, I have seen it later. I haven't changed it, but another mistake in the path.

Comment (moylop260) [inline git/README.md]: Is theee a mistake in the path in the doc or your had a mistake in your local path?

File: git/README.md (hunk 5 above)
Comment (pedrobaeza) [inline git/README.md]: Check by yourself the change: https://github.com/OCA/maintainer-quality-tools/commit/a55cdc5e115917973730dea75bfc1adef7bf7a04

Comment (moylop260) [inline git/README.md]: I got it
Thank you

**Pull request merged**